compiler/z/codegen/S390Peephole.cpp

```diff
@@ -1282,10 +1282,8 @@ TR_S390PostRAPeephole::LGFRReduction()
 bool
 TR_S390PostRAPeephole::ConditionalBranchReduction(TR::InstOpCode::Mnemonic branchOPReplacement)
    {
-   bool disabled = comp()->getOption(TR_DisableZ13) || comp()->getOption(TR_DisableZ13LoadImmediateOnCond);
-
    // This optimization relies on hardware instructions introduced in z13
-   if (!comp()->target().cpu.getSupportsArch(TR::CPU::z13) || disabled)
+   if (!TR::Compiler->target.cpu.getSupportsArch(TR::CPU::z13))
```

Comment:
Is it intentional that we won't return false here (i.e. not perform the optimization that relies on z13) if the disableZ13 option was specified by the user ?

Reply:
Is the user has specified `disableZ13` option then `getSupportsArch(TR::CPU::z13)` will return false, meaning we would enter the `if` statement and return false, meaning we won't do the reduction. This all looks correct.